File: library/src/main/java/com/uploadcare/android/library/upload/Uploader.java
package com.uploadcare.android.library.upload;

import com.uploadcare.android.library.api.UploadcareFile;
import com.uploadcare.android.library.callbacks.UploadcareFileCallback;
import com.uploadcare.android.library.exceptions.UploadFailureException;

public interface Uploader {

    UploadcareFile upload() throws UploadFailureException;

    void uploadAsync(UploadcareFileCallback callback) throws UploadFailureException;

    Uploader store(boolean store);
}

File: library/src/main/java/com/uploadcare/android/library/upload/Uploader.kt
package com.uploadcare.android.library.upload

import com.uploadcare.android.library.api.UploadcareFile
import com.uploadcare.android.library.callbacks.ProgressCallback
import com.uploadcare.android.library.callbacks.UploadFileCallback
import com.uploadcare.android.library.exceptions.UploadFailureException

interface Uploader {

    @Throws(UploadFailureException::class)
    fun upload(progressCallback: ProgressCallback? = null): UploadcareFile

    fun uploadAsync(callback: UploadFileCallback)

    fun cancel()

    fun store(store: Boolean): Uploader
}